src/language_server.ts

```diff
@@ -78,6 +83,85 @@ export class LanguageServer {
 	}
 
 
+	/**
+	 * Returns an array of string(s) representing the version of the Inmanta Language Server
+	 * that needs to be installed. The function checks if the environment variable INMANTA_LS_PATH is
+	 * set and uses it to install the LS from the specified path in editable mode. If the environment variable is not set,
+	 * it checks for the presence of requirements.txt file and installs the LS from it.
+	 * If requirements.txt file is not present, it installs the latest version of Inmanta Language Server.
+	 *
+	 * @returns {string[]} An array of string(s) representing the version of the Inmanta Language Server to be installed.
+	 */
+	languageServerVersionToInstall(): string[] {
+		const version = [];
+
+		if (process.env.INMANTA_LS_PATH) {
+		  version.push("-e", process.env.INMANTA_LS_PATH);
+		  log(`Installing Language Server from local source "${process.env.INMANTA_LS_PATH}"`);
+		} else {
+		  // Check for the presence of requirements.txt
+		  if (fs.existsSync(REQUIREMENTS_PATH)) {
+			version.push("-r", REQUIREMENTS_PATH);
+			log(`Installing Language Server from requirements file "${REQUIREMENTS_PATH}"`);
+		  } else {
+			version.push("inmantals");
+		  }
+		}
+
+		return version;
+	}
+
+	/**
+	 * Returns the version of the installed Inmanta Language Server, or null if it's not installed.
+	 *
+	 * @returns {string | null} The version of the installed Inmanta Language Server, or null if it's not installed.
+	 */
+	getInstalledInmantaLSVersion(): string | null {
+		try {
+		  const output = cp.execSync(`${this.pythonPath} -m pip show inmantals`).toString();
+		  const versionMatch = output.match(/^Version: (.+)$/m);
+		  if (versionMatch) {
+			return versionMatch[1];
+		  }
+		} catch (error) {}
+		return null;
+	}
+
+	/**
+	 * Checks if the correct version of the Inmanta Language Server is installed.
+	 *
+	 * @returns {boolean} True if the correct version of the Inmanta Language Server is installed, false otherwise.
+	 */
+	isCorrectInmantaLSVersionInstalled(): boolean {
+		// The LS is installed in editable mode via the env var
+		if (process.env.INMANTA_LS_PATH) {
+			return true;
+		}
+		// No requirements specified
+		if (!fs.existsSync(REQUIREMENTS_PATH)) {
```

Comment:
Just as a sanity check, this would never occur in practice, would it?

Reply:
in production it should never happen. But in dev there is no requirements.txt file as it will only be added by the ci/cd.

Reply:
Right.